File: algorithms-and-data-structure-for-beginners/2 - static-arrays/leet-code-26-remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize) {
  const int OFFSET = 100; // porque os numeros vão de -100 a 100, não posso deixar índices negativos
  const int TAMANHO_INTERVALO = 201; // de -100 a 100 temos 201 números possíveis

  int k = 0;
  int numsQueJaApareceram[TAMANHO_INTERVALO] = {};

  for (int i = 0; i < numsSize; i++) {
    if (numsQueJaApareceram[nums[i]+OFFSET] != 1) {
      nums[k] = nums[i];
      k++;
      numsQueJaApareceram[nums[i]+OFFSET] = 1;
    }
  }

  return k;
}

File: algorithms-and-data-structure-for-beginners/2 - static-arrays/leet-code-26-remove-duplicates-from-sorted-array.go
package staticarrays

func removeDuplicates(nums []int) int {
	n := len(nums)

	left, right := 0, 0

	for right < n {
		nums[left] = nums[right]
		for right < n && nums[left] == nums[right] {
			right++
		}
		left++
	}

	return left
}

func removeDuplicates2(nums []int) int {
	left := 1

	for right := 1; right < len(nums); right++ {
		if nums[right] != nums[right-1] {
			nums[left] = nums[right]
			left++
		}
	}

	return left
}
